portal-api/src/v1/gef/services/clone-deal.service.test.ts

```diff
@@ -0,0 +1,210 @@
+        userId: maker._id,
+        bank: mockBank,
+        auditDetails: mockAuditDetails,
+      }),
+    ).rejects.toThrow(new InvalidDealIdError(invalidDealId));
+  });
+
+  it('throws InvalidParameterError when the bank id is not a string', async () => {
+    // Arrange
+    const bankId = 123;
+
+    // Act & Assert
+    await expect(() =>
+      cloneDealToLatestVersion({
+        dealId: existingDealId,
+        bankInternalRefName,
+        additionalRefName,
+        maker,
+        userId: maker._id,
+        bank: { id: bankId } as unknown as Bank,
+        auditDetails: mockAuditDetails,
+      }),
+    ).rejects.toThrow(new InvalidParameterError('bank.id', bankId));
+  });
+
+  it('throws DealNotFoundError when the dealId is valid but does not correspond to a deal', async () => {
+    // Arrange
+    findOneMock.mockResolvedValueOnce(null);
+
+    // Act & Assert
+    await expect(() =>
+      cloneDealToLatestVersion({
+        dealId: existingDealId,
+        bankInternalRefName,
+        additionalRefName,
+        maker,
+        userId: maker._id,
+        bank: mockBank,
+        auditDetails: mockAuditDetails,
+      }),
+    ).rejects.toThrow(new DealNotFoundError(existingDealId));
+  });
+
+  it('gets existing deal', async () => {
+    // Act
+    await cloneDealToLatestVersion({
+      dealId: existingDealId,
+      bankInternalRefName,
+      additionalRefName,
+      maker,
+      userId: maker._id,
+      bank: mockBank,
+      auditDetails: mockAuditDetails,
+    });
+
+    // Assert
+    expect(findOneMock).toHaveBeenCalledTimes(1);
+    expect(findOneMock).toHaveBeenCalledWith({ _id: { $eq: new ObjectId(existingDealId) }, 'bank.id': { $eq: mockBank.id } });
+  });
+
+  it('inserts a cloned deal', async () => {
+    // Act
+    await cloneDealToLatestVersion({
+      dealId: existingDealId,
+      bankInternalRefName,
+      additionalRefName,
+      maker,
+      userId: maker._id,
+      bank: mockBank,
+      auditDetails: mockAuditDetails,
+    });
+
+    // Assert
+    expect(insertOneMock).toHaveBeenCalledTimes(1);
+    expect(insertOneMock).toHaveBeenCalledWith({
+      _id: expect.any(ObjectId) as ObjectId,
+      dealType: existingDeal.dealType,
+      version: getCurrentGefDealVersion(),
+      createdAt: Date.now(),
+      updatedAt: Date.now(),
+      facilitiesUpdated: Date.now(),
+      eligibility: {
+        ...mockLatestEligibilityCriteria,
+        updatedAt: Date.now(),
+      },
+      status: DEAL_STATUS.DRAFT,
+      submissionType: null,
+      submissionDate: null,
+      submissionCount: 0,
+      bankInternalRefName,
+      additionalRefName,
+      maker,
+      bank: mockBank,
+      ukefDealId: null,
+      checkerId: null,
+      editedBy: [maker._id],
+      exporter: cloneExporter(existingDeal.exporter),
+      portalActivities: [],
+      supportingInformation: {},
+      clonedDealId: existingDealId,
+      auditRecord: generateAuditDatabaseRecordFromAuditDetails(mockAuditDetails),
```

Comment:
Can we use a fixture and then assert only the expected changed properties?

Reply:
Similiar to above, about 25 properties are changed